src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_time_stm32f4_nucleo_wifi_sntp.c

```diff
@@ -50,17 +46,27 @@ static const char SNTP_REQUEST[SNTP_MSG_SIZE] = {
 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
 0xd5, 0x22, 0x0e, 0x35, 0xb8, 0x76, 0xab, 0xea};
 
+extern void HAL_Delay(uint32_t delay_ms);
 static WiFi_Status_t    sntp_start( char* sntp_server, uint32_t sntp_port, uint8_t* sock_id );
 static WiFi_Status_t    sntp_send_request( uint8_t sock_id );
 static sntp_status_t    sntp_await_response( uint8_t sock_id );
 static sntp_response_t* sntp_malloc_response( void );
 static int32_t          sntp_parse_response( char* response );
 static void             sntp_free_response( sntp_response_t** r );
 static WiFi_Status_t    sntp_stop( uint8_t sock_id );
-
 static uint32_t sntp_ntohl( uint32_t n );
 
 /**
+ * @function
+ * @brief Returns epoch time (seconds since 1970/01/01 00:00:00)
+ * @retval epoch time as an int32_t
+ */
+posix_time_t xi_bsp_time_sntp_getseconds_posix( void )
```

Comment:
posix_time_t needs to be int32_t (the 4 bytes that will be copied from the SNTP response), but xi_time_t seems to be declared as long. Not a problem for this 32-bit micro, but I'm not sure it will work if we ever re-use this code for a 64-bit architecture. What do you think? Depending on the endianness we may have to re-implement ntoh() anyway, so I'm not sure what to think?